File: CrewControl/src/app/models/Person.ts
import { IPerson } from "../interfaces/person";

export class Person implements IPerson {
    id: number;
    firstName: string;
    lastName: string;
    age: number;
    gender: string;
    maritalStatus: string;
    address: string;
    city: string;
    state: string;
    country: string;
    email: string;
    phoneNumber: string;
    dateOfBirth: Date;
  
    constructor(id: number , firstName: string, lastName: string, age: number, gender: string, maritalStatus: string,
       address: string, city: string, state: string, country: string, email: string, phoneNumber: string, dateOfBirth: string) {
      this.id = id || 0;
      this.firstName = firstName;
      this.lastName = lastName;
      this.age = age;
      this.gender = gender;
      this.maritalStatus = maritalStatus;
      this.address = address;
      this.city = city;
      this.state = state;
      this.country = country;
      this.email = email;
      this.phoneNumber = phoneNumber;
      this.dateOfBirth = new Date(dateOfBirth);
    }
    }

File: CrewControl.NetCoreApi/Model/Person.cs
#pragma warning disable CA1050 // Declare types in namespaces
public class Person
#pragma warning restore CA1050 // Declare types in namespaces
{
    public int Id { get; set; }
    public string FirstName { get; set; }
    public string? LastName { get; set; }
    public int?  Age { get; set; }
    public string? Address { get; set; }
    public string? Gender { get; set; }
    public string? MaritalStatus { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Country { get; set; }
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public DateTime? DateOfBirth { get; set; }
}